### ISSUE TITLE
Ensure that all functions that call bus.Dispatch are methods

### ISSUE THREAD

**Issue by zserge** [grafana/grafana#44135]
As part of #44130 we need to prepare all the bus.Dispatch callers to have some kind of a context - all such functions must belong to a struct, which could inject the callees.

As usual, AWK comes to the rescue estimating the scope of the issue (`dispatch.awk`):

```awk
/testing\.T/ { nextfile }
/^func.*$/ { f = $0; }
/\.Dispatch/ { if (f !~ /func \(/) print(FILENAME"\t"f); }
```

Applying this script to the OSS code base results in the following functions:

```go
api/admin.go	func AdminGetStats(c *models.ReqContext) response.Response {
api/admin_users.go	func AdminUpdateUserPassword(c *models.ReqContext) response.Response {
api/admin_users.go	func AdminDeleteUser(c *models.ReqContext) response.Response {
api/admin_users.go	func AdminEnableUser(c *models.ReqContext) response.Response {
api/alerting.go	func ValidateOrgAlert(c *models.ReqContext) {
api/alerting.go	func GetAlertStatesForDashboard(c *models.ReqContext) response.Response {
api/alerting.go	func GetAlerts(c *models.ReqContext) response.Response {
api/alerting.go	func GetAlert(c *models.ReqContext) response.Response {
api/alerting.go	func getAlertNotificationsInternal(c *models.ReqContext) ([]*models.AlertNotification, error) {
api/alerting.go	func GetAlertNotificationByID(c *models.ReqContext) response.Response {
api/alerting.go	func GetAlertNotificationByUID(c *models.ReqContext) response.Response {
api/alerting.go	func CreateAlertNotification(c *models.ReqContext) response.Response {
api/alerting.go	func DeleteAlertNotification(c *models.ReqContext) response.Response {
api/alerting.go	func DeleteAlertNotificationByUID(c *models.ReqContext) response.Response {
api/alerting.go	func NotificationTest(c *models.ReqContext) response.Response {
api/alerting.go	func PauseAlert(c *models.ReqContext) response.Response {
api/alerting.go	func PauseAllAlerts(c *models.ReqContext) response.Response {
api/apikey.go	func DeleteAPIKey(c *models.ReqContext) response.Response {
api/dashboard.go	func isDashboardStarredByUser(c *models.ReqContext, dashID int64) (bool, error) {
api/dashboard.go	func getUserLogin(ctx context.Context, userID int64) string {
api/dashboard.go	func getDashboardHelper(ctx context.Context, orgID int64, id int64, uid string) (*models.Dashboard, response.Response) {
api/dashboard.go	func GetDashboardVersions(c *models.ReqContext) response.Response {
api/dashboard.go	func GetDashboardVersion(c *models.ReqContext) response.Response {
api/dashboard.go	func GetDashboardTags(c *models.ReqContext) {
api/dashboard_snapshot.go	func CreateDashboardSnapshot(c *models.ReqContext) response.Response {
api/dashboard_snapshot.go	func GetDashboardSnapshot(c *models.ReqContext) response.Response {
api/dashboard_snapshot.go	func DeleteDashboardSnapshotByDeleteKey(c *models.ReqContext) response.Response {
api/dashboard_snapshot.go	func DeleteDashboardSnapshot(c *models.ReqContext) response.Response {
api/dashboard_snapshot.go	func SearchDashboardSnapshots(c *models.ReqContext) response.Response {
api/datasources.go	func AddDataSource(c *models.ReqContext) response.Response {
api/datasources.go	func getRawDataSourceById(ctx context.Context, id int64, orgID int64) (*models.DataSource, error) {
api/datasources.go	func getRawDataSourceByUID(ctx context.Context, uid string, orgID int64) (*models.DataSource, error) {
api/datasources.go	func GetDataSourceByName(c *models.ReqContext) response.Response {
api/datasources.go	func GetDataSourceIdByName(c *models.ReqContext) response.Response {
api/login_oauth.go	func syncUser(
api/org_invite.go	func GetPendingOrgInvites(c *models.ReqContext) response.Response {
api/org_invite.go	func AddOrgInvite(c *models.ReqContext) response.Response {
api/org_invite.go	func inviteExistingUserToOrg(c *models.ReqContext, user *models.User, inviteDto *dtos.AddInviteForm) response.Response {
api/org_invite.go	func GetInviteInfoByCode(c *models.ReqContext) response.Response {
api/org_invite.go	func updateTempUserStatus(ctx context.Context, code string, status models.TempUserStatus) (bool, response.Response) {
api/org_invite.go	func applyUserInvite(ctx context.Context, user *models.User, invite *models.TempUserDTO, setActive bool) (bool, response.Response) {
api/password.go	func SendResetPasswordEmail(c *models.ReqContext) response.Response {
api/password.go	func ResetPassword(c *models.ReqContext) response.Response {
api/playlist.go	func ValidateOrgPlaylist(c *models.ReqContext) {
api/playlist.go	func SearchPlaylists(c *models.ReqContext) response.Response {
api/playlist.go	func GetPlaylist(c *models.ReqContext) response.Response {
api/playlist.go	func LoadPlaylistItems(ctx context.Context, id int64) ([]models.PlaylistItem, error) {
api/playlist.go	func DeletePlaylist(c *models.ReqContext) response.Response {
api/playlist.go	func CreatePlaylist(c *models.ReqContext) response.Response {
api/playlist.go	func UpdatePlaylist(c *models.ReqContext) response.Response {
api/playlist_play.go	func populateDashboardsByID(ctx context.Context, dashboardByIDs []int64, dashboardIDOrder map[int64]int) (dtos.PlaylistDashboardsSlice, error) {
api/playlist_play.go	func populateDashboardsByTag(ctx context.Context, orgID int64, signedInUser *models.SignedInUser, dashboardByTag []string, dashboardTagOrder map[string]int) dtos.PlaylistDashboardsSlice {
api/pluginproxy/pluginproxy.go	func NewApiPluginProxy(ctx *models.ReqContext, proxyPath string, route *plugins.Route,
api/preferences.go	func SetHomeDashboard(c *models.ReqContext) response.Response {
api/quota.go	func GetUserQuotas(c *models.ReqContext) response.Response {
api/quota.go	func UpdateUserQuota(c *models.ReqContext) response.Response {
api/search.go	func Search(c *models.ReqContext) response.Response {
api/signup.go	func SignUp(c *models.ReqContext) response.Response {
api/signup.go	func verifyUserSignUpEmail(ctx context.Context, email string, code string) (bool, response.Response) {
api/stars.go	func StarDashboard(c *models.ReqContext) response.Response {
api/stars.go	func UnstarDashboard(c *models.ReqContext) response.Response {
api/user.go	func GetUserByLoginOrEmail(c *models.ReqContext) response.Response {
api/user.go	func UpdateUserActiveOrg(c *models.ReqContext) response.Response {
api/user.go	func handleUpdateUser(ctx context.Context, cmd models.UpdateUserCommand) response.Response {
api/user.go	func getUserTeamList(ctx context.Context, orgID int64, userID int64) response.Response {
api/user.go	func getUserOrgList(ctx context.Context, userID int64) response.Response {
api/user.go	func validateUsingOrg(ctx context.Context, userID int64, orgID int64) bool {
api/user.go	func UserSetUsingOrg(c *models.ReqContext) response.Response {
api/user.go	func ChangeUserPassword(c *models.ReqContext) response.Response {
api/user.go	func SetHelpFlag(c *models.ReqContext) response.Response {
api/user.go	func ClearHelpFlags(c *models.ReqContext) response.Response {
cmd/grafana-cli/commands/reset_password_command.go	func resetPasswordCommand(c utils.CommandLine, sqlStore *sqlstore.SQLStore) error {
components/dashdiffs/compare.go	func CalculateDiff(ctx context.Context, options *Options) (*Result, error) {
login/brute_force_login_protection.go	func validatePasswordSet(password string) error {
login/grafana_login.go	func validatePasswordSet(password string) error {
login/ldap_login.go	func validatePasswordSet(password string) error {
login/ldap_login.go	func DisableExternalUser(ctx context.Context, username string) error {
middleware/org_redirect.go	func OrgRedirect(cfg *setting.Cfg) web.Handler {
services/alerting/rule.go	func getAlertNotificationUIDByIDAndOrgID(ctx context.Context, notificationID int64, orgID int64) (string, error) {
services/dashboards/folder_service.go	func getFolder(ctx context.Context, query models.GetDashboardQuery) (*models.Dashboard, error) {
services/login/loginservice/loginservice.go	func updateUser(ctx context.Context, user *models.User, extUser *models.ExternalUserInfo) error {
services/login/loginservice/loginservice.go	func updateUserAuth(ctx context.Context, user *models.User, extUser *models.ExternalUserInfo) error {
services/login/loginservice/loginservice.go	func syncOrgRoles(ctx context.Context, user *models.User, extUser *models.ExternalUserInfo) error {
services/ngalert/notifier/alertmanager.go	func init() {
services/provisioning/dashboards/file_reader.go	func getOrCreateFolderID(ctx context.Context, cfg *config, service dashboards.DashboardProvisioningService, folderName string) (int64, error) {
services/provisioning/utils/utils.go	func CheckOrgExists(ctx context.Context, orgID int64) error {
services/search/service.go	func setStarredDashboards(ctx context.Context, userID int64, hits []*Hit) error {
```

Or, if you want a list of files:

- [ ] api/admin.go
- [ ] api/admin_users.go
- [ ] api/alerting.go @zserge 
- [ ] api/apikey.go
- [ ] api/dashboard.go @yangkb09 
- [ ] api/dashboard_snapshot.go @yangkb09 
- [ ] api/datasources.go @idafurjes 
- [ ] api/login_oauth.go
- [ ] api/org_invite.go https://github.com/grafana/grafana/pull/44530
- [ ] api/password.go @yangkb09 
- [x] api/playlist.go https://github.com/grafana/grafana/pull/44485
- [x] api/playlist_play.go https://github.com/grafana/grafana/pull/44485
- [ ] api/pluginproxy/pluginproxy.go @zserge 
- [ ] api/plugins.go @idafurjes 
- [ ] api/preferences.go @idafurjes 
- [ ] api/quota.go @idafurjes 
- [ ] api/search.go @ying-jeanne 
- [x] api/signup.go https://github.com/grafana/grafana/pull/44485
- [ ] api/stars.go @ying-jeanne 
- [x] api/team.go https://github.com/grafana/grafana/pull/44218
- [x] api/user.go https://github.com/grafana/grafana/pull/44468
- [ ] cmd/grafana-cli/commands/reset_password_command.go
- [ ] components/dashdiffs/compare.go
- [ ] login/brute_force_login_protection.go
- [ ] login/grafana_login.go
- [ ] login/ldap_login.go
- [ ] middleware/org_redirect.go
- [ ] services/alerting/rule.go
- [ ] services/dashboards/folder_service.go
- [ ] services/login/loginservice/loginservice.go
- [ ] services/ngalert/notifier/alertmanager.go
- [ ] services/provisioning/dashboards/file_reader.go
- [ ] services/provisioning/utils/utils.go
- [ ] services/search/service.go

Basically, what's needed is to replace `func Foo()` with `func (obj *someObj) Foo()`, depending on the Foo better belongs. In the `api` package that would be `HTTPServer`. Feel free to leave comments if some functions can not be assigned to any of the structs.

**Comment by marefr** [comment]
I removed this from Backend Platform Backlog project - it's a project we don't want to use any longer and planning to archive.

**Comment by ying-jeanne** [comment]
Design doc: https://docs.google.com/document/d/1s3lXFQbRHjmxviWSX7XH1HhPRvHiu_b7nIivY7iuZCg/edit#
